Document BalEnv async interop use case (#190)
ballerina-platform/ballerina-platform.github.io: learn/calling-java-code-from-ballerina.md

As you can see, you need to first construct an instance of the `ArrayDeque` class. The `arrayDequeObj` variable refers to an `ArrayDeque` object. Then, you need to pass this variable to both the `pop` and `push` functions because the corresponding Java methods are instance methods of the`ArrayDeque` class. Therefore, you need an instance of the `ArrayDeque` class in order to invoke its instance methods. You can think of the `arrayDequeObj` variable as the method receiver.

#### Calling Methods Asynchronously

> Ballerina internally uses a fixed number of threads, therefore when calling Java method, it should return in a reasonable time frame in order to avoid starvation in Ballerina code execution.

Reviewer: ```suggestion
Ballerina internally uses a fixed number of threads. Therefore, when calling a Java method, it should return in a reasonable time frame in order to avoid starvation in the Ballerina code execution.
```

> If the given Java method executes a time consuming (ie: blocking) task such as IO operation, better to do that in a separate thread while yielding the original thread to continue the Ballerina code execution.

Reviewer: ```suggestion
If the given Java method executes a time consuming (i.e., blocking) task such as an IO operation, better to do that in a separate thread while yielding the original thread to continue the Ballerina code execution.
```

> In this case, Ballerina Scheduler needs to be informed that the work is being completed asynchronously by invoking `markAsync` method in `BalEnv` object. When the work is completed, `complete` method has to be called with the return value. Note that the original return value is ignored.

Reviewer: ```suggestion
In this case, Ballerina Scheduler needs to be informed that the work is being completed asynchronously by invoking the `markAsync` method in the `BalEnv` object. When the work is completed, the `complete` method has to be called with the return value. 

>**Note:** The original return value is ignored.
```
Better to add the markdown syntax for Notes.
